cl/search/tasks.py

```diff
@@ -684,6 +684,19 @@ def update_children_docs_by_query(
         if not parent_instance:
             return
         count_query = Position.objects.filter(person_id=parent_instance_id)
+    elif (
+        es_document is ESRECAPDocument
+        and event_table == EventTable.DOCKET_ENTRY
```

Comment:
This feels a little weird. What happens if we don't both with this `and` statement and just let people send in `ESRECAPDocument`'s?

Reply:
Yes, I've refactored this to ensure it only operates with the expected arguments. `ESRECAPDocument` requires `event_table` to be `EventTable.DOCKET_ENTRY` for updating the DocketEntry's RECAPDocuments, or `event_table` to be either `None` or `EventTable.DOCKET` for updating Docket's RECAPDocuments. The task will be aborted if `event_table` is `EventTable.RECAP_DOCUMENT`.